ics-frontend/src/app/app-routing.module.ts

```diff
@@ -1,7 +1,15 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import {GalleryComponent} from './gallery/gallery.component';
+import { SingleImagePageComponent } from './single-image-page/single-image-page.component';
+import { ImageClassificationComponent } from './image-classification/image-classification.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: 'gallery', component: GalleryComponent},
+  {path: 'image-classification', component: ImageClassificationComponent},
+  {path: 'image-classification/:id', component: SingleImagePageComponent},
+  {path: '', redirectTo: 'image-classification', pathMatch: 'full'}
```

Comment:
You'll still need to have a home page for the initial search

Reply:
@asalov my home page would be the image-classification - it is the place the user can paste the link, from there on we would get image-classification/:id with specific classified image;

alternatively the user would go to /gallery and then select an image and again get the /image-classification/:id to view the tagging of a particular image